cumulusci/tasks/salesforce/users/tests/test_photos.py

```diff
@@ -0,0 +1,46 @@
+import pytest  # noqa: F401
+from unittest import mock
+from cumulusci.tasks.salesforce.tests.util import create_task
+
+from cumulusci.tasks.salesforce.users import UploadProfilePhoto  # noqa: F401
+from cumulusci.core.exceptions import CumulusCIException  # noqa: F401
+
+
+class TestUploadProfilePhoto:
+    def setup_method(self):
+        self.task = create_task(
+            UploadProfilePhoto, {"photo_path": "path/to/profile/photo.png"}
+        )
+
+    def test_get_user_id(self):
+        expected = "user_id_0001"
+
+        self.task.sf = mock.Mock()
+        self.task.sf.restful.return_value = {"identity": expected}
+        self.task.logger = mock.Mock()
+
+        actual = self.task.get_user_id()
+
+        assert expected == actual
+
+        self.task.sf.restful.assert_called_once_with("")
+        self.task.logger.info.assert_called_once_with(
+            f"Uploading profile photo for the default User with ID {expected}."
+        )
+
+    @mock.patch("builtins.open", new_callable=mock.mock_open, read_data="photo fi")
+    @mock.patch("base64.b64encode")
+    @mock.patch("os.path")
+    @mock.patch("json.loads")
+    def test_run_task__failed_inserting_content_version(
+        self, open_mock, base64_base64encode_mock, os_path_mock, json_loads_mock
+    ):
+        # Calculated variables
+        path = self.task.options["photo_path"]
+
+        # Mocks
+        self.task.get_user_id = mock.Mock()
+        self.task.logger = mock.Mock()
+        self.task.logger.info._expected_calls = [
+            mock.call(f"Setting user photo to {path}")
+        ]
```

Comment:
I would be inclined to write this as an integration test that uses a real file on disk and asserts that the expected ContentVersion data was prepared, rather than a strict unit test. I try to reserve use of mocks for cases where the tests need to interact with an external service or would otherwise be slow.

Reply:
Will do.   I haven't touched the test since the re-factor.

Reply:
The tests are integration tests in regards to files.  `pathlib` is not mocked.  Only simple salesforce is mocked.